### PR TITLE
Fix iosxr discard_changes netconf rpc issue

### PR DESCRIPTION
**Pull request by pabelanger**
##### SUMMARY
Fix iosxr discard_changes netconf rpc issue

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
netconf_rpc

### CODE REVIEW

File: lib/ansible/plugins/netconf/__init__.py [hunk 1]
```diff
@@ -223,9 +223,14 @@ def dispatch(self, rpc_command=None, source=None, filter=None):
         """
         if rpc_command is None:
             raise ValueError('rpc_command value must be provided')
+        req = fromstring(rpc_command)
+        resp = self.m.dispatch(req, source=source, filter=filter)
 
-        resp = self.m.dispatch(fromstring(rpc_command), source=source, filter=filter)
-        return resp.data_xml if hasattr(resp, 'data_xml') else resp.xml
+        result = resp.xml
```
Comment (ganeshrn) [inline lib/ansible/plugins/netconf/__init__.py]: better to move this line in else part. something like
```
        if hasattr(resp, 'data_xml') or hasattr(resp, 'data_ele'):
            result = resp.data_xml
        else:
            result = resp.xml
```

Comment (pabelanger) [reply lib/ansible/plugins/netconf/__init__.py]: sure, why better? Should be the same output right?

Comment (ganeshrn) [reply lib/ansible/plugins/netconf/__init__.py]: In most cases data_xml object is present the resp object, returning `xml` object is catch all type of scenario, hence in former code result variable is assigned twice most of the times.

**Comment by pabelanger** [comment]
Pull-request updated, HEAD is now 61dd89e240bc13c95699b0a3f3b9c6028f41f3ba

**Review by ganeshrn — COMMENTED** [review]
Please refer the changes to dispatch function [here](https://gist.github.com/ganeshrn/dc51151251de09d6193dbc95cb9663d1)

**Pull request closed without merging**